INI.sublime-syntax

```diff
@@ -2,182 +2,182 @@
-                    - match: (,)\s*$\n?
-                      captures:
-                        1: punctuation.separator.sequence.ini
-                      set: maybe-value
-                    - include: string
-                    - match: $\n?
-                      pop: true
-        - match: '(\[)(\d+)(\])'
+            - mapping-value
+            - mapping-separator
+        - match: (\[)(\d+)(\])
           captures:
-             1: punctuation.section.brackets.begin.ini
-             2: constant.numeric.integer.decimal.ini
-             3: punctuation.section.brackets.end.ini
+            1: punctuation.section.brackets.begin.ini
+            2: constant.numeric.integer.decimal.ini
+            3: punctuation.section.brackets.end.ini
         - match: \,
           scope: punctuation.separator.sequence.ini
         - include: string
-        - match: $\n?
-          pop: true
+        - include: eol-pop
 
-  maybe-value:
-    - meta_content_scope: meta.mapping.value.ini
-    - match: '^(?=\S)'
+  mapping-separator:
+    - match: '{{separator}}'
+      scope: punctuation.separator.mapping.key-value.ini
       pop: true
-    - match: '(?=.+{{separator}})'
+
+  mapping-value:
+    - meta_content_scope: meta.mapping.ini
+    - match: \-(?=\s*($|#))
+      scope: meta.mapping.value.ini keyword.operator.arithmetic.ini
       pop: true
-    - include: number
-    - include: constant
+    - match: (?=\S)
+      set: maybe-value-firstline
+    - match: \n
+      set: maybe-value-multiline
+
+  maybe-value-firstline:
+    - meta_scope: meta.mapping.value.ini
     - match: (,)\s*$\n?
       captures:
         1: punctuation.separator.sequence.ini
+      set: maybe-value-multiline
+    - include: constant
+    - include: number
     - include: string
+    - include: eol-pop
 
-  number:
-    - match: '[+-]?\b\d*(\.)\d+(?:(?:E|e)[+-]?\d+)?(F|f)?\b'
-      scope: constant.numeric.float.decimal.ini
+  maybe-value-multiline:
+    - meta_content_scope: meta.mapping.value.ini
+    - include: comment
+    - match: ^(?=\S)
+      pop: true
+    - match: (?=.+{{separator}})
+      pop: true
+    - match: (,)\s*$\n?
       captures:
-        1: punctuation.separator.decimal.ini
-        2: storage.type.numeric.ini
-    - match: '[+-]?\b\d+\b'
-      scope: constant.numeric.integer.decimal.ini
-
-  hex-number:
-    - match: \b\h+\b
-      scope: constant.numeric.integer.hexadecimal.ini
+        1: punctuation.separator.sequence.ini
+    - include: constant
+    - include: number
+    - include: string
 
   constant:
     - match: \b(?i:true|false)\b
       scope: constant.language.ini
-    - match: '\bdword\b'
+    - match: \bdword\b
       scope: storage.type.dword.ini
       set:
         - meta_content_scope: meta.mapping.value.ini
         - match: ':'
           scope: punctuation.separator.colon.ini
           set:
-            - meta_content_scope: meta.mapping.value.ini
+            - meta_scope: meta.mapping.value.ini
```

Comment:
Could you revert this to `meta_content_scope` to prevent doubling of `meta.mapping.value.ini` for the semicolon?
After this I think it's perfect and ready to merge.

Reply:
Another residence from version-2 experiments. Took another approach to solve the issue and to avoid putting meta.mapping.value into any of those sub contexts, though.